compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -2188,17 +2188,36 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
           isJSFunctionDef(currentClassSym)) {
         val flags = js.MemberFlags.empty.withNamespace(namespace)
         val body = {
+          def genAsUnaryOp(op: js.UnaryOp.Code): js.Tree =
+            js.UnaryOp(op, genThis())
+          def genAsBinaryOp(op: js.BinaryOp.Code): js.Tree =
+            js.BinaryOp(op, genThis(), jsParams.head.ref)
+
           if (currentClassSym.get == HackedStringClass) {
             /* Hijack the bodies of String.length and String.charAt and replace
              * them with String_length and String_charAt operations, respectively.
              */
             methodName.name match {
```

Comment:
Maybe it's a bit odd. I guess we do it this way because we have the `Symbol`s of the classes anyway, and comparing them is a reference equality test. For the methods, we would have to look up all their symbols only to successfully compare them once, so it would be a lot of wasted effort.